components/glean/build.gradle

```diff
@@ -0,0 +1,201 @@
+// =============================================
+// We're doing something a bit weird here, so
+// READ THIS CAREFULLY before editing this file.
+// =============================================
+//
+// This is a lightly-modified version of the `build.gradle` file from Glean,
+// which you can find at `../external/glean/glean-core/android/build.gradle`.
```

Comment:
I don't like the fact that this has copied a lot of logic from the Glean build.gradle, seems like it could easily get out of sync. But, I also don't have a better idea. We could possibly consider trying to move some of this logic into the upstream build.gradle, somehow...?

Reply:
I'm not sure what "upstream" you refer to here, and without looking at the differences carefully I'm not sure how practical this is, but working out how to split this and use `apply from:` seems worthwhile - it scares me a little that the README above says "update `./build.gradle` to track any build changes made in Glean" and then discover you mean track stuff in this 200 line semi-complicated file and where it's not at all obvious how it relates to glean's original.

(I don't have a concrete ask here, just expressing a hope that @badboy might have ideas here :)

Reply:
Right, I meant "upstream" as in "into some gradle files that live in the Glean repo and are maintained by the Glean team" in a very abstract sense